### PR TITLE
chore: Relax protobuf constraints

### LINKED ISSUE #3101: Upgrade protobuf past version 4
**Is your feature request related to a problem? Please describe.**

Feast pins protobuf to `<4`. This means that for M1 Apple machines, we have to employ some hacks. It would be nice to not have to do that. Also maintaining an upper bound on a popular library means that installing Feast alongside other libraries involves friction.

**Describe the solution you'd like**

Ideally, we remove the upper bound for protobuf in `setup.py`. It seems like the breaking changes are pretty minor: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates

**Describe alternatives you've considered**

N/A

### PR DESCRIPTION
**Pull request by kevjumba**
Signed-off-by: Kevin Zhang <kzhang@tecton.ai>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3101

### CODE REVIEW

File: sdk/python/feast/proto_json.py [hunk 1]
```diff
@@ -111,7 +111,16 @@ def from_json_object(
...
             )
 
-    _patch_proto_json_encoding(Value, to_json_object, from_json_object)
+    def from_json_object_updated(
+        parser: _Parser, value: JsonObject, message: ProtoMessage, path: str
+    ):
+        from_json_object(parser, value, message)
+
+    current_version = pkg_resources.get_distribution("protobuf").version
+    if version.parse(current_version) < version.parse("3.20"):
```
Comment (achals) [inline sdk/python/feast/proto_json.py:120]:
```suggestion
    current_protobuf_version = pkg_resources.get_distribution("protobuf").version
    if version.parse(current_protobuf_version) < version.parse("3.20"):
```

Comment (achals) [reply sdk/python/feast/proto_json.py:120]: Maybe link to any relevant issue here?

File: sdk/python/feast/proto_json.py [hunk 2]
```diff
@@ -141,14 +150,27 @@ def _patch_feast_repeated_value_json_encoding():
...
-    _patch_proto_json_encoding(RepeatedValue, to_json_object, from_json_object)
+    def from_json_object(
+        parser: _Parser, value: JsonObject, message: ProtoMessage
+    ) -> None:
+        array = value if isinstance(value, list) else value["val"]
+        for item in array:
+            parser.ConvertMessage(item, message.val.add())
+
+    current_version = pkg_resources.get_distribution("protobuf").version
+    if version.parse(current_version) < version.parse("3.20"):
```
Comment (achals) [inline sdk/python/feast/proto_json.py:168]: same comment

**Review by achals — APPROVED** [review]

**Review by achals — APPROVED** [review]
/lgtm

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**